src/main/java/com/squareup/catalog/demo/example/ApplyTaxToAllIItemsExample.java

```diff
@@ -0,0 +1,194 @@
+        return null;
+      }
+
+      // Append the new taxes to the complete list of taxes.
+      allTaxes.addAll(listResponse.getObjects());
+
+      // Move to the next page.
+      cursor = listResponse.getCursor();
+    } while (cursor != null);
+
+    return allTaxes;
+  }
+
+  /**
+   * Displays all taxes in the console so the user can select one. The selected tax is returned.
+   *
+   * @param allTaxes a complete list of taxes
+   * @return the selected tax
+   */
+  private CatalogObject promptUserForTax(List<CatalogObject> allTaxes) throws ApiException {
+    // Display the list of taxes.
+    logger.info("Available Taxes:");
+    for (int i = 0; i < allTaxes.size(); i++) {
+      CatalogTax tax = allTaxes.get(i).getTaxData();
+      logger.info(i + ". " + tax.getName() + " (" + tax.getPercentage() + "%)");
+    }
+
+    // Prompt the user to select a tax.
+    String userTaxIndex = promptUserInput("Select a tax to apply to all items: ");
+    int taxIndex = -1;
+    try {
+      taxIndex = Integer.parseInt(userTaxIndex);
+    } catch (NumberFormatException e) {
+      logger.error("Invalid tax index specified");
+      return null;
+    }
+
+    if (taxIndex < 0 || taxIndex >= allTaxes.size()) {
+      logger.error("Specified index is out of range.");
+    }
+
+    return allTaxes.get(taxIndex);
+  }
+
+  /**
+   * Iterates over all items, applying the specified tax to each.
+   *
+   * @param tax the tax to apply to all items
+   */
+  private void applyTaxToAllItems(CatalogApi catalogApi, CatalogObject tax) throws ApiException {
+    logger.info("Applying " + tax.getTaxData().getName() + " to all items");
+
+    final String taxId = tax.getId();
+    String cursor = null;
+    int totalItemsApplied = 0;
+    int totalItemsVisited = 0;
+    final long startTimeMillis = System.currentTimeMillis();
+    do {
+      // Retrieve a page of items.
+      ListCatalogResponse listResponse = catalogApi.listCatalog(cursor, "ITEM");
+      if (checkAndLogErrors(listResponse.getErrors())) {
+        return;
+      }
+
+      List<CatalogObject> items = listResponse.getObjects();
+      if (items.size() == 0) {
+        if (cursor == null) {
+          logger.info("No items found in catalog.");
+          return;
+        }
+      } else {
+        // Figure out which items to apply the tax to.
+        UpdateItemTaxesRequest updateItemTaxesRequest = new UpdateItemTaxesRequest()
+            .addTaxesToEnableItem(taxId);
+        for (CatalogObject item : items) {
+          // Ignore non-regular items and items already linked to the tax.
+          CatalogItem.ProductTypeEnum itemType = item.getItemData().getProductType();
+          if ((itemType == null || itemType == REGULAR) &&
+              !item.getItemData().getTaxIds().contains(taxId)) {
+            updateItemTaxesRequest.addItemIdsItem(item.getId());
+          }
+        }
+
+        // Add the tax to the items.
+        if (updateItemTaxesRequest.getItemIds().size() > 0) {
```

Comment:
it's better to have the example handling large batch in pages. it needs to make sure on size of items doesn't exceed the max item number in a batch.